Upgrade to rel imx 5.4.70 2.3.0

MaxKrummenacher opened the pull request:
@otavio @zandrey related to #622

Update to NXP relase 5.4.70_2.3.0

With an fslc-xxx distro and use-nxp-bsp, in order to use the NXP forks of gstreamer and the NXP provided i.MX specific plugins imx-gst1.0-plugin one needs the following:

```
PREFERRED_VERSION_gstreamer1.0 = "1.16.imx"
PREFERRED_VERSION_gstreamer1.0-plugins-base = "1.16.imx"
PREFERRED_VERSION_gstreamer1.0-plugins-good = "1.16.imx"
PREFERRED_VERSION_gstreamer1.0-plugins-bad = "1.16.imx"
```

With the above above change to the configuration I was able to build core-image-weston plus some tools and:

- boot core-image-weston
- start weston-launch
- run weston-simple-egl and glmark2-es2-wayland
- play a video file with gstreamer

on a i.MX 6Q and i.MX 8M Mini based machine.
Note that I did not use the bootloader though.

Used build configuration:
```
Build Configuration:
BB_VERSION           = "1.49.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "arm-fslc-linux-gnueabi"
MACHINE              = "apalis-imx6"
DISTRO               = "fslc-xwayland"
DISTRO_VERSION       = "3.3-snapshot-20210130"
TUNE_FEATURES        = "arm vfp cortexa9 neon thumb callconvention-hard"
TARGET_FPU           = "hard"
meta                 
meta-poky            = "master:45c02843d181b5f1872ad30f4d847c395233a676"
meta-oe              
meta-multimedia      
meta-python          = "HEAD:29059f9c8f36c0915d7a432f45f7766af61ad5c9"
meta-freescale       = "upgrade-to-rel_imx_5.4.70_2.3.0:96fde216285500e0c223922fb740130a694e568d"
meta-freescale-3rdparty = "master:95904deaaea96aa40c5539b590639ed21ceaf3e0"
meta-freescale-distro = "master:e83f54d0b763826f076e8aed1fc16dbf5b2739e5"
```

otavio commented on recipes-graphics/drm/libdrm_2.4.102.imx.bb:
```diff
@@ -69,5 +68,3 @@ RRECOMMENDS_${PN}-drivers_append_imxgpu = " ${PN}-vivante"
 PACKAGECONFIG_append_imxgpu = " vivante"
 PACKAGECONFIG[vivante] = "-Dvivante=true,-Dvivante=false"
-
-PACKAGE_ARCH = "${MACHINE_SOCARCH}"
```
Why you are dropping this? This can cause task checksum changes if we use non-i.MX machines or use mainline specific machines.

MaxKrummenacher replied:
I removed it because:
If I put `PREFERRED_VERSION_libdrm = "2.4.102.imx"` in local.conf and `MACHINE=qemuarm bitbake libdrm` the build fails because MACHINE_SOCARCH is not defined.

```
ERROR: libdrm-2.4.102.imx-r0 do_write_config: Execution of '/build/krm/fsl-community-bsp/build/tmp/work/${MACHINE_SOCARCH}-fslc-linux-gnueabi/libdrm/2.4.102.imx-r0/temp/run.do_write_config.58725' failed with exit
 code 1:
/build/krm/fsl-community-bsp/build/tmp/work/${MACHINE_SOCARCH}-fslc-linux-gnueabi/libdrm/2.4.102.imx-r0/temp/run.do_write_config.58725: line 150: /build/krm/fsl-community-bsp/build/tmp/work/-fslc-linux-gnueabi/li
bdrm/2.4.102.imx-r0/meson.cross: No such file or directory
WARNING: /build/krm/fsl-community-bsp/build/tmp/work/${MACHINE_SOCARCH}-fslc-linux-gnueabi/libdrm/2.4.102.imx-r0/temp/run.do_write_config.58725:150 exit 1 from 'cat > /build/krm/fsl-community-bsp/build/tmp/work/$
{MACHINE_SOCARCH}-fslc-linux-gnueabi/libdrm/2.4.102.imx-r0/meson.cross  <<EOF
...
```
And, since libdrm is in the conf/machine/include/imx-base.inc I thought for the machines including that incfile PACKAGE_ARCH is set magically in the classes/fsl-dynamic-packagearch.bbclass.

But looking at it more closely there is an issue. PREFERRED_VERSION_mxX and the MACHINE_SOCARCH_FILTER_append_imxgpu use different overrides. So e.g. i.MX 6ULL will behave badly.
I will change the PREFERRED_VERSION to read PREFERRED_VERSION_libdrm_imxgpu ?= "2.4.102.imx".

Should I instead put a `COMPATIBLE_MACHINE = "(imxgpu)"`?

MaxKrummenacher replied:
Probably better to keep `PACKAGE_ARCH = "${MACHINE_SOCARCH}"` and add `COMPATIBLE_MACHINE = "(imxgpu)"`
Then we don't change the libdrm version used for i.MX machines not falling under `_imxgpu`.

otavio replied:
Perfect.

otavio reviewed (changes requested):
I'd like to thank you both for this hard work and say this is really appreciated. Couple of things I noticed:

1. there are 3 or 4 patches which has duplicated signed off by entires for @zandrey, please amend it to have just one
2. the libdrm change seems wrong.

MaxKrummenacher commented:
Changes in 0d5a118..65f478f:

Rebased on top of current master
Removed duplicated Signed-off in `a9b65168 imx-test: upgrade to release 5.4.70_2.3.0`
Kept `PACKAGE_ARCH = "${MACHINE_SOCARCH}"` and added `COMPATIBLE_MACHINE = "(mx6|mx7|mx8)"` to libdrm.
